components/core/src/package/archive.rs

```diff
@@ -417,6 +419,77 @@ pub trait FromArchive: Sized {
     fn from_archive(archive: &mut PackageArchive) -> result::Result<Self, Self::Error>;
 }
 
+// Exposes the extended archive and header metadata
+#[derive(Deserialize, Serialize)]
+pub struct PackageArchiveInfo {
+    pub format_version:    String,
+    pub key_name:          String,
+    pub hash_type:         String,
+    pub signature_raw:     String,
+    pub origin:            String,
+    pub name:              String,
+    pub version:           Option<String>,
+    pub release:           Option<String>,
+    pub checksum:          Option<String>,
+    pub target:            PackageTarget,
+    pub is_a_service:      bool,
+    pub deps:              Vec<PackageIdent>,
+    pub tdeps:             Vec<PackageIdent>,
+    pub build_deps:        Vec<PackageIdent>,
+    pub build_tdeps:       Vec<PackageIdent>,
+    pub exposes:           Vec<u16>,
+    pub pkg_services:      Vec<PackageIdent>,
+    pub resolved_services: Vec<PackageIdent>,
+    pub manifest:          Option<String>,
+    pub config:            Option<String>,
+    pub svc_user:          Option<String>,
+    pub ld_run_path:       Option<String>,
+    pub ldflags:           Option<String>,
+    pub cflags:            Option<String>,
+}
+
+impl PackageArchiveInfo {
+    pub fn new(path: impl Into<PathBuf>) -> Result<Self> {
+        let src = path.into();
+        let mut archive = PackageArchive::new(src.clone())?;
```

Comment:
It might be better to have the main work of this function be in something like 

```rust
PackageArchiveInfo::new(archive: &mut PackageArchive) -> Result<PackageArchiveInfo>
```
with 

```rust
PackageArchiveInfo::from_path(path: impl Into<PathBuf>) -> Result<PackageArchiveInfo>
```
calling _that_.